clients/src/main/java/org/apache/kafka/common/requests/FetchResponse.java

```diff
@@ -277,14 +278,14 @@ public PartitionData(Errors error,
                              long highWatermark,
                              long lastStableOffset,
                              long logStartOffset,
-                             Integer preferredReadReplica,
+                             Optional<Integer> preferredReadReplica,
                              List<AbortedTransaction> abortedTransactions,
                              T records) {
             this.error = error;
             this.highWatermark = highWatermark;
             this.lastStableOffset = lastStableOffset;
             this.logStartOffset = logStartOffset;
-            this.preferredReadReplica = Optional.ofNullable(preferredReadReplica);
+            this.preferredReadReplica = preferredReadReplica;
```

Comment:
Do you think it's worth checking that the id is not negative, if provided?

Reply:
Doesn't seem like we normally do this kind of validation. If a negative was given, I suppose we'd end up reverting to the leader since that Node wouldn't be found.